src/modules/frost/bench_impl.h

```diff
@@ -0,0 +1,218 @@
+
+        /* TODO: FIXME - here the same experiment is run (count) times, and nonces are reused.
+         * We do not want to generate new nonces, otherwise the signing time is compromised. */
+        for (j = 0; j < BENCH_FROST_THR_PARTICIPANTS; j++) {
+            data->state[i].nonces[j]->used = 0;
+        }
+        CHECK(secp256k1_frost_sign(data->ctx,
+                                   &signature_share,
+                                   data->msgs[i],
+                                   BENCH_FROST_THR_PARTICIPANTS,
+                                   &data->state[i].keypairs[signer_index],
+                                   data->state[i].nonces[signer_index],
+                                   data->state[i].signing_commitments) == 1);
+    }
+}
+
+static void bench_frost_aggregate(void* arg, int iters) {
+    bench_frost_data *data = (bench_frost_data *)arg;
+    int i, signer_index;
+    unsigned char signature[65];
+
+    for (i = 0; i < iters; i++) {
+        signer_index = i % BENCH_FROST_THR_PARTICIPANTS;
+        /* We aggregate 3 shares out of the 3 provided */
+        CHECK(secp256k1_frost_aggregate(data->ctx,
+                                        signature,
+                                        data->msgs[i],
+                                        &data->state[i].keypairs[signer_index],
+                                        data->state[i].public_keys,
+                                        data->state[i].signing_commitments,
+                                        data->state[i].signature_shares,
+                                        BENCH_FROST_THR_PARTICIPANTS) == 1);
+    }
+}
+
+static void bench_frost_verify(void* arg, int iters) {
+    bench_frost_data *data = (bench_frost_data *)arg;
+    int i;
+
+    for (i = 0; i < iters; i++) {
+        CHECK(secp256k1_frost_verify(data->ctx,
+                                     data->state[i].signature,
+                                     data->msgs[i],
+                                     data->state[i].public_keys) == 1);
+    }
+}
+
+static void bench_frost_data_init(bench_frost_data *data, int iters) {
+    int i, j;
+    unsigned char hiding_seed[32], binding_seed[32];
+    secp256k1_frost_vss_commitments *_dealer_commitments;
+    secp256k1_frost_keygen_secret_share _shares_by_participant[BENCH_FROST_MAX_PARTICIPANTS];
+
+    /* Allocate benchmark data */
+    data->ctx = secp256k1_context_create(SECP256K1_CONTEXT_NONE);
+    data->state = (secp256k1_frost_data_per_iter *)malloc(iters * sizeof(secp256k1_frost_data_per_iter));
+    data->msgs = (const unsigned char **)malloc(iters * sizeof(unsigned char *));
+
+    /*
+     * Prepare data for each iteration: at each iteration, we use different
+     * key-pairs, nonces, and commitments. In this way, each iteration performs
+     * equivalent (but not the same) operations.
+     */
+    for (i = 0; i < iters; i++) {
+        uint32_t MSGLEN=32;
```

Comment:
This declaration was temporarily inserted while removing the MSGLEN parameter. It compiled under amd64, but breaks under different platforms. It needs to be removed.